### ISSUE TITLE
[no-extra-semi] False negative with class property

### ISSUE THREAD

**Issue by BBosman** [typescript-eslint/typescript-eslint#1167]
**Repro**

The code below fails to trigger a violation on `no-extra-semi`.

```JSON
{
  "rules": {
    "no-extra-semi": "error"
  }
}
```

```TS
export class TestBinding {
  public $scope?: IScope = void 0;;
}
```

**Expected Result**
To flag the double semicolon as a violation of `no-extra-semi`.

**Actual Result**
It does not trigger a violation.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.6.0` |
| `@typescript-eslint/parser`        | `2.6.0` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `10.15.0` |
| `npm`                              | `6.12.0` |
